### PR TITLE
Make legacy tests work on Python 3

### PR DESCRIPTION
**Pull request by willbarton**
This is a cleanup of map/filter generators and translating unittest `assertItemsEqual` to `assertCountEqual`.

### CODE REVIEW

File: cfgov/legacy/housing_counselor/cleaner.py [hunk 1]
```diff
@@ -71,9 +71,9 @@ def title_case(s):
     )
 
     parts[0] = parts[0].title()
-    parts = map(
+    parts = list(map(
         lambda part: part.title() if part not in lower_case else part,
         parts
-    )
+    ))
```
Comment (chosak) [inline cfgov/legacy/housing_counselor/cleaner.py]: Is this change needed as the resulting map `parts` is joined just below?

Comment (willbarton) [reply cfgov/legacy/housing_counselor/cleaner.py]: Oh, no, I don't think so.
